### ISSUE TITLE
Is it possible to use chassis for production?

### ISSUE THREAD

**Issue by tareiking** [Chassis/Chassis#123]
Just curious whether there are any security/performance or other nuances to using chassis as a production ready stack?

I'd had success with using the Vagrant Digital Ocean plugin to push to a live droplet, but I'm curious what might be needed to 'harden' chassis for production; or is this just pie-in-the-sky thinking?

**Comment by Japh** [comment]
Chassis seems to me to be configured explicitly for a development environment. I'm not sure using Chassis for production can even make sense.

I'd be inclined to setup a post-receive hook for your project within Chassis to deploy changes to production. Or maybe even a simple rsync to get everything up there!

There're probably better ways, but those are simple.

Are you really asking about the OS level stuff? Cause I think it'd be pretty doable to tweak the provisioning scripts Chassis uses to allow a production push of some kind for the entire environment.

**Comment by rmccue** [comment]
> Just curious whether there are any security/performance or other nuances to using chassis as a production ready stack?

There's a bunch of stuff in Chassis that assume it's a local/development install, and although I can't think of anything that'd be insecure, I'm not willing to trust that. :)

> Are you really asking about the OS level stuff? Cause I think it'd be pretty doable to tweak the provisioning scripts Chassis uses to allow a production push of some kind for the entire environment.

Realistically, we need to do this. Separate the core Puppet of Chassis and make that available as a separate project which can be used to provision sites.

**Comment by tareiking** [comment]
@japh You are right with your comments about syncing / post-receive hooks and there are definitely other platforms for deployment. 

> Realistically, we need to do this. Separate the core Puppet of Chassis and make that available as a separate project which can be used to provision sites.

Likely out-of-scope for chassis, but answers my question. While using the same puppet manifest's to provision a new production environment would be nice, similar provisioning tools are already freely available as stand-alone projects.

TY guys, closing...

**Comment by rmccue** [comment]
I'm going to reopen, because we should actually do it. :)

**Comment by thespotvin** [comment]
I'd love to have an automated staging/production deployment method built in as an option, but in the meantime does anyone have any resources for setting up a staging/production workflow around chassis?

**Comment by kambanwait** [comment]
+1 

Looking at how I can use this locally and then deploy to production for new and existing wordpress site's I have.

**Comment by dtbaker** [comment]
Just digging up this old issue, does anyone have a preferred workflow for getting stuff to staging/production?

I feel like I'm re-inventing the wheel here with some custom scripts, it would be great to see how others are handling it (especially if anyone has had a play with mergebot or similar to sync stage/prod back to dev). 

Thanks!

**Comment by matgargano** [comment]
@dtbaker  did you ever get anywhere ? I have a chassis setup that I need to at least put on a staging server because the prod version of my app lives on VIP

**Comment by dtbaker** [comment]
nah man, went about this project a completely different way with github + CI + cloudformation and staging/prod environments. I might have a look back at chassis again for the next project.
